src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -21,77 +42,698 @@ const ActorInit En_Goroiwa_InitVars = {
+        effectPos.z = ((Math_Rand_ZeroOne() * 50.0f) * temp_f24) * Math_Sins(angle2);
+        effectVelocity.x = effectPos.x * 0.2f;
+        effectVelocity.y = (Math_Rand_ZeroOne() * 15.0f) + 2.0f;
+        effectVelocity.z = effectPos.z * 0.2f;
+        Math_Vec3f_Sum(&effectPos, thisPos, &effectPos);
+        func_80029E8C(globalCtx, &effectPos, &effectVelocity, &effectPos, -340, 33, 28, 2, 0,
+                      (Math_Rand_ZeroOne() * 7.0f) + 1.0f, 1, 0, 70, -1, 1, D_0400D340);
+    }
+
+    effectPos.x = thisPos->x;
+    effectPos.y = thisPos->y + colliderHeightOffset[temp_v0];
+    effectPos.z = thisPos->z;
+    func_80033480(globalCtx, &effectPos, 80.0f, 5, 70, 110, 1);
+    func_80033480(globalCtx, &effectPos, 90.0f, 5, 110, 160, 1);
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, 64676, ICHAIN_CONTINUE), ICHAIN_F32_DIV1000(minVelocityY, 50536, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),   ICHAIN_F32(uncullZoneForward, 1500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 150, ICHAIN_CONTINUE),   ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Goroiwa/EnGoroiwa_Draw.s")
+void EnGoroiwa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static f32 D_80A4DF10[] = { 0.0f, 595.0f };
+
+    EnGoroiwa* this = THIS;
+    s32 pathIdx;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    func_80A4BD04(this, globalCtx);
+    pathIdx = this->actor.params & 0xFF;
+    if (pathIdx == 0xFF) {
+        // Translation: Error: Invalid arg_data
+        osSyncPrintf("Ｅｒｒｏｒ : arg_data が不正(%s %d)(arg_data 0x%04x)\n", "../z_en_gr.c", 1033,
+                     this->actor.params);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if (globalCtx->setupPathList[pathIdx].count < 2) {
+        // Translation: Error: Invalid Path Data
+        osSyncPrintf("Ｅｒｒｏｒ : レールデータ が不正(%s %d)\n", "../z_en_gr.c", 1043);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    func_80061ED4(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    ActorShape_Init(&this->actor.shape, D_80A4DF10[(this->actor.params >> 10) & 1], ActorShadow_DrawFunc_Circle, 9.4f);
+    this->actor.shape.unk_14 = 200;
+    EnGoroiwa_SetSpeed(this, globalCtx);
+    func_80A4C188(this, globalCtx);
+    func_80A4C1C4(this, globalCtx, 0);
+    func_80A4C264(this);
+    func_80A4BE54(this, globalCtx);
+    func_80A4D5E0(this);
+    // Translation: (Goroiwa)
+    osSyncPrintf("(ごろ岩)(arg 0x%04x)(rail %d)(end %d)(bgc %d)(hit %d)\n", this->actor.params,
+                 this->actor.params & 0xFF, (this->actor.params >> 8) & 3, (this->actor.params >> 10) & 1,
+                 this->actor.initPosRot.rot.z & 1);
+}
+
+void EnGoroiwa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Collider_DestroyJntSph(globalCtx, &THIS->collider);
+}
```

Comment:
can a `this = THIS` temp be afforded at all?

Reply:
Nope.